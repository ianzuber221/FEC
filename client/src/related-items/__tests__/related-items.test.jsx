/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, cleanup } from '@testing-library/react';
import RelatedItems from '../related-items';
import ProductList from '../ProductList';
import AddOutfit from '../addOutfit';
import Comparison from '../comparison';
import Product from '../product';

// import  from '../';

const testCard = {
  id: 1,
  image:
    'https://ngca.net/wp-content/uploads/2020/09/image-coming-soon-placeholder.png',
  category: 'COSTCO CHIC',
  name: 'Costco Shirt',
  price: '$5',
  discountedPrice: null,
  description: 'Cool shirt from Costco',
  features: 'It is what it is',
};

describe('Related-items tests', () => {
  test('Related-items renders correctly', () => {
    const tree = renderer.create(<RelatedItems />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Related-items renders correctly', () => {
    const tree = renderer.create(<Comparison />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Product renders correctly', () => {
    const tree = renderer.create(<Product />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render product', () => {
    const { getbyTestId } = render(<Product/>);
    const star = findByTestId("Star");
    expect(star).not.toBe('undefined');
  });
});
