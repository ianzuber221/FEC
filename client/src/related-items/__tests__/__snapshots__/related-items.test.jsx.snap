// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related-items tests Related-items renders correctly 1`] = `
<div
  className="sc-eCYdqJ bIZnIq"
>
  <h2>
    Related Items
  </h2>
  <div
    className="sc-bczRLJ jIHupg"
  >
    <button
      className="slider leftSlider"
      onClick={[Function]}
      type="button"
    />
    <div
      id="relatedList"
      onScroll={[Function]}
    />
    <button
      className="slider rightSlider"
      onClick={[Function]}
      type="button"
    >
      〉
    </button>
  </div>
  <h2>
    Your Outfits
  </h2>
  <div
    className="sc-bczRLJ jIHupg"
  >
    <button
      className="slider leftSlider"
      onClick={[Function]}
      type="button"
    />
    <div
      className="sc-gsnTZi cwXtxL"
    >
      <div
        id="addOutfitContainer"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,11 L22.1551134,17.4623449 C22.6217314,17.7592836 23,18.4433532 23,19.0093689 L23,20.9906311 C23,21.5480902 22.5605417,22 21.9975383,22 L2.00246167,22 C1.44881738,22 1,21.5566468 1,20.9906311 L1,19.0093689 C1,18.4519098 1.3786449,17.7590442 1.84488659,17.4623449 L12,11 Z M15,5 C15,3.34314575 13.6568542,2 12,2 C10.3431458,2 9,3.34314575 9,5 C9,5.93157601 9.41137234,6.80169553 10.0908534,7.31422922 C11,8 12,8 12,9.5 L12,11"
            fill="none"
            stroke="#000"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
        <div>
          Add to Outfit
        </div>
      </div>
    </div>
    <div
      id="outfitList"
      onScroll={[Function]}
    />
    <button
      className="slider rightSlider"
      onClick={[Function]}
      type="button"
    >
      〉
    </button>
  </div>
</div>
`;

exports[`Related-items tests Related-items renders correctly 2`] = `
<div
  className="sc-bczRLJ jIHupg"
>
  <button
    className="slider leftSlider"
    onClick={[Function]}
    type="button"
  />
  <div
    id="relatedList"
    onScroll={[Function]}
  />
  <button
    className="slider rightSlider"
    onClick={[Function]}
    type="button"
  >
    〉
  </button>
</div>
`;

exports[`Related-items tests Related-items renders correctly 3`] = `
<div
  className="sc-gsnTZi cwXtxL"
>
  <div
    id="addOutfitContainer"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12,11 L22.1551134,17.4623449 C22.6217314,17.7592836 23,18.4433532 23,19.0093689 L23,20.9906311 C23,21.5480902 22.5605417,22 21.9975383,22 L2.00246167,22 C1.44881738,22 1,21.5566468 1,20.9906311 L1,19.0093689 C1,18.4519098 1.3786449,17.7590442 1.84488659,17.4623449 L12,11 Z M15,5 C15,3.34314575 13.6568542,2 12,2 C10.3431458,2 9,3.34314575 9,5 C9,5.93157601 9.41137234,6.80169553 10.0908534,7.31422922 C11,8 12,8 12,9.5 L12,11"
        fill="none"
        stroke="#000"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
      />
    </svg>
    <div>
      Add to Outfit
    </div>
  </div>
</div>
`;

exports[`Related-items tests Related-items renders correctly 4`] = `
<div
  className="sc-bczRLJ jIHupg"
>
  <button
    className="slider leftSlider"
    onClick={[Function]}
    type="button"
  />
  <div
    className="sc-gsnTZi cwXtxL"
  >
    <div
      id="addOutfitContainer"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12,11 L22.1551134,17.4623449 C22.6217314,17.7592836 23,18.4433532 23,19.0093689 L23,20.9906311 C23,21.5480902 22.5605417,22 21.9975383,22 L2.00246167,22 C1.44881738,22 1,21.5566468 1,20.9906311 L1,19.0093689 C1,18.4519098 1.3786449,17.7590442 1.84488659,17.4623449 L12,11 Z M15,5 C15,3.34314575 13.6568542,2 12,2 C10.3431458,2 9,3.34314575 9,5 C9,5.93157601 9.41137234,6.80169553 10.0908534,7.31422922 C11,8 12,8 12,9.5 L12,11"
          fill="none"
          stroke="#000"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
        />
      </svg>
      <div>
        Add to Outfit
      </div>
    </div>
  </div>
  <div
    id="outfitList"
    onScroll={[Function]}
  />
  <button
    className="slider rightSlider"
    onClick={[Function]}
    type="button"
  >
    〉
  </button>
</div>
`;
